src/table/virtualized-table/Wrapper.tsx

```diff
@@ -25,8 +24,15 @@ const Wrapper = (props: WrapperProps) => {
     [pageSize, page]
   );
 
-  // Reset page on new layout
-  useOnPropsChange(() => setPage(0), [layout]);
+  useOnPropsChange(() => {
```

Comment:
If that is the case then I think we should, so that they behave the same

Reply:
I have another PR that is based on this PR. Where I have move and restructured the page logic. It will make that change in that PR instead. Ok?

Reply:
So if I understand you correctly. If the data size changes, essentionally `qSize.qcy or qSize.qcx` changes. The pagination table should reset to page `0`?  I'm not seeing that behaviour or that kind of logic in the code

Reply:
I was suspected that there might be bug with `useOnPropsChange` regarding using referencing object values as deps, and i think i was not mistaken. 

I think it should be fine with primitive data types but as you can see for object values => if you change something deep inside, it will always detect `hasChanged` as true!

<img width="412" alt="Screenshot 2023-03-06 at 09 53 27" src="https://user-images.githubusercontent.com/29652890/223062424-b80b491e-63e9-4aa4-ad5d-b56b75e6f901.png">

Reply:
You're right, the logic isn't like that. If you end up on an empty page, you go back to the first one. So it is just a simpler version of what you have here. When needed, you should run setPage(0)

Reply:
and yes leave it to the next pr

Reply:
@a-m-dev In your example code, you assign `newProp` with a new object? In which case it should detect that it has changed

Reply:
Yes, I mean if something has changes in layout, you will receive a new entire layout object and if that was not the case, still the result will be the same while something changes inside of same object 👇


<img width="417" alt="Screenshot 2023-03-06 at 10 42 08" src="https://user-images.githubusercontent.com/29652890/223073688-dc794784-798e-4b33-b765-c08392d697b3.png">

Reply:
This is still the correct behavior? `oldProp[0] === newProp[0]; // Are not the same`.

So I've created some unit tests for`useOnPropsChange` locally and it behaves as I would expect. So mayby I'm missing your point?

Reply:
That is correct, my point is in above example, if you update the qcy with **1**, it still going to detect as object has  changed. 
(sorry for wrong example 😅)

<img width="409" alt="Screenshot 2023-03-06 at 10 56 48" src="https://user-images.githubusercontent.com/29652890/223077206-d24d1d1a-151a-45d9-95e5-d6aa28e91634.png">

Reply:
That is what I want. If the property is a new layout it should trigger the callback function. There are cases where we could say pass in `layout.qHyperCube.qSize.qcy` directly as a dependency instead, but it tends to be hard to know which layout change should trigger a change or not. So I mostly just use `layout` directly to be on the safe side.